src/python/grpcio_tests/tests/unit/_metadata_flags_test.py

```
@@ -0,0 +1,244 @@
+
+
+def perform_stream_unary_future(channel, wait_for_ready=None):
+    channel.stream_unary(_STREAM_UNARY).future(
+        iter([_REQUEST] * test_constants.STREAM_LENGTH),
+        timeout=test_constants.LONG_TIMEOUT,
+        wait_for_ready=wait_for_ready).result(
+            timeout=test_constants.LONG_TIMEOUT)
+
+
+def perform_stream_stream_call(channel, wait_for_ready=None):
+    response_iterator = channel.stream_stream(_STREAM_STREAM).__call__(
+        iter([_REQUEST] * test_constants.STREAM_LENGTH),
+        timeout=test_constants.LONG_TIMEOUT,
+        wait_for_ready=wait_for_ready)
+    for _ in response_iterator:
+        pass
+
+
+_ALL_CALL_CASES = [
+    perform_unary_unary_call, perform_unary_unary_with_call,
+    perform_unary_unary_future, perform_unary_stream_call,
+    perform_stream_unary_call, perform_stream_unary_with_call,
+    perform_stream_unary_future, perform_stream_stream_call
+]
+
+
+class MetadataFlagsTest(unittest.TestCase):
+
+    def check_connection_does_failfast(self, fn, channel, wait_for_ready=None):
+        try:
+            fn(channel, wait_for_ready)
+        except BaseException as e:  # pylint: disable=broad-except
+            self.assertIn('StatusCode.UNAVAILABLE', str(e))
+
+    def test_call_wait_for_ready_default(self):
+        for perform_call in _ALL_CALL_CASES:
+            self.check_connection_does_failfast(perform_call,
+                                                create_dummy_channel())
+
+    def test_call_wait_for_ready_disabled(self):
+        for perform_call in _ALL_CALL_CASES:
+            self.check_connection_does_failfast(
+                perform_call, create_dummy_channel(), wait_for_ready=False)
+
+    def test_call_wait_for_ready_enabled(self):
+        # To test the wait mechanism, Python thread is required to make
+        #   client set up first without handling them case by case.
+        # Also, Python thread don't pass the unhandled exceptions to
```

Will it be a problem for the end user if these exceptions are never surfaced to their application?

---

The threading logic here is for this unit test case specifically. User have the freedom to create their own exception handling logic.